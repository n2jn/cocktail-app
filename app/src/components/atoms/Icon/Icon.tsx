import {View} from 'react-native';
import FeatherIcon from 'react-native-vector-icons/Feather';
import {IconButtonProps as FeatherIconButtonProps} from 'react-native-vector-icons/Icon';
import {scale} from '~gStyles/size';

type AcceptableIconSize = 8 | 16 | 24 | 32 | 64;

type IconProps = Omit<FeatherIconButtonProps, 'size'> & {
  size: AcceptableIconSize;
};

export const Icon = ({size, onPress, ...iconprops}: IconProps) => {
  return (
    <FeatherIcon.Button
      disabled={!onPress}
      onPress={onPress}
      size={scale(size)}
      {...iconprops}
    />
  );
};
