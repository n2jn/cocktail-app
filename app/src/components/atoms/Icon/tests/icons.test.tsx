import renderer from 'react-test-renderer';
import {Icon} from '../Icon';

jest.mock('react-native-vector-icons/Feather');
test('Should render correctly', () => {
  const tree = renderer.create(<Icon size={16} name={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});
