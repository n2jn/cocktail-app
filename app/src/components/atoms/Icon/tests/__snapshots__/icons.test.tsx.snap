// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Should render correctly 1`] = `
<View
  disabled={true}
  name=""
  size={38}
/>
`;

exports[`Should render correctly 1`] = `
<View
  disabled={true}
  name=""
  size={38}
/>
`;
